routers/web/repo/setting.go

```diff
@@ -341,6 +342,12 @@ func SettingsPost(ctx *context.Context) {
 			return
 		}
 
+		if form.PushMirrorUsePublicKey && (form.PushMirrorUsername != "" || form.PushMirrorPassword != "") {
```

Comment:
Okay, now I'm certainly confused: You can't enter either username or password if the checkbox is selected.
Okay, fine.
But does this indeed mean that Gitea will generate a new SSH key to use?
If that's the case, we should probably
1. notify the user with a description/ tooltip
2. disable and clear the username/ password fields when the `SSH` checkbox has been clicked.

Reply:
> But does this indeed mean that Gitea will generate a new SSH key to use?

Yes, it's purely to prevent users to upload their personal SSH keys.